src/main/groovy/nextflow/executor/OarExecutor.groovy

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2013-2018, Centre for Genomic Regulation (CRG).
+ * Copyright (c) 2013-2018, Paolo Di Tommaso and the respective authors.
+ *
+ *   This file is part of 'Nextflow'.
+ *
+ *   Nextflow is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation, either version 3 of the License, or
+ *   (at your option) any later version.
+ *
+ *   Nextflow is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with Nextflow.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package nextflow.executor
+import java.nio.file.Path
+
+import groovy.util.logging.Slf4j
+import nextflow.processor.TaskRun
+
+
+/**
+ * Implements a executor for OAR cluster
+ *
+ * See http://oar.imag.fr
+ */
+@Slf4j
+class OarExecutor extends AbstractGridExecutor {
+
+    /**
+     * Gets the directives to submit the specified task to the cluster for execution
+     *
+     * @param task A {@link TaskRun} to be submitted
+     * @param result The {@link List} instance to which add the job directives
+     * @return A {@link List} containing all directive tokens and values.
+     */
+
+    protected List<String> getDirectives( TaskRun task, List<String> result ) {
+        assert result !=null
+
+        result << '-n' << getJobNameFor(task)
+        result << '-d' << quote(task.workDir)
+
+        // the requested queue name
+        if( task.config.queue ) {
+            result << '-q'  << (String)task.config.queue
+        }
+
+        // miscellaneous cluster options semicollon separated
+        if( task.config.clusterOptions ) {
+          for (String item : task.config.clusterOptions.toString().tokenize(';')) {
+            result << item << ''
+          }
+        }
+
+      return result
+
+    }
+
+    @Override
+    String getHeaders( TaskRun task ) {
+        String result = super.getHeaders(task)
+        result += "cd ${quote(task.workDir)}\n"
+        return result
+    }
```

Comment:
Is this really necessary if in the directives is added `-d`, that is supposed to set the job working directory?